documentation/upgrading_to_bolt_3.md

```diff
@@ -0,0 +1,83 @@
+# Upgrading to Bolt 3.0
+
+Welcome to Bolt 3.0! 
+
+**Before you upgrade:**
+- Take a look at the
+  [Changelog](https://github.com/puppetlabs/bolt/blob/main/CHANGELOG.md).
+
+You can view upgrade instructions for your operating system at [Installing
+Bolt](bolt_installing.md).
+
+## Migrating configuration files
+
+Bolt 3.0 introduces changes to Bolt's configuration file layout and content, including
+removing `bolt.yaml` and renaming several configuration options. Here's how to transition to the new
+world order.
+
+## Automated migration
+
+For project-level configuration files you can update to use the latest Bolt best practices using our
+migration command:
+
+_\*nix shell command_
+
+```shell
+bolt project migrate
+```
+
+_PowerShell cmdlet_
+
+```powershell
+Update-BoltProject
+```
+
+## Manual migration
+
+For user- and system-level configuration files, or if you're not comfortable having Bolt re-write
+your project-level configuration files, follow these steps to migrate to the new file layout.
+
+### User- and system-level config
+
+1. Move any [transport configuration](bolt_transports_reference.md) to be under the
+   `inventory-config` key, like so:
+
+   ```
+   inventory-config:
+       ssh:
+         password: hunter2!
+         user: bolt
+       winrm:
+         password: hunter2!
+         user: bolt
+   ```
+   Transport configuration keys are `ssh`, `winrm`, `pcp`, `local`, `docker`, `remote`, and
+   `transport`.
+1. Rename the following configuration options:
+    - `plugin_hooks` to `plugin-hooks`
+1. Rename `apply_settings` to `apply_settings` and move the configuration to `bolt-project.yaml`.
```

Comment:
```suggestion
1. Rename `apply_settings` to `apply_settings` and move the configuration to `bolt-project.yaml`
   in your project directory.
```

Just so users don't think this means to create a `bolt-project.yaml` in the user/system config directories. 😬